src-tauri/src/main.rs

```diff
@@ -60,6 +60,8 @@ async fn farming(path: String, reward_address: String, plot_size: u64) -> bool {
             // if there is an error, restart another farmer, and start listening again, in a loop
             loop {
                 let result = error_receiver.recv().await;
+                // probably node restarted, so wait 7 seconds before starting farmer
+                thread::sleep(Duration::from_millis(7000));
                 match result {
                     // we have received an error, let's restart the farmer
```

Comment:
Why it won't restart? In case of error, it will send an error using the mpsc::channel (farmer.rs -> line 69). 
And it will restart a new farmer, in case the previous farmer is crashed.

Reply:
Because there is `unwrap()` below, If you unwrap `Err` task will exit (because tokio handles panics for you and I think will print a message as well, but not more than that).

Reply:
Inside the farmer code, we have this:
```rust
tokio::spawn(async move {
        let result = multi_farming.wait().await;
        match result {
            Err(error) => {
                log::error!("{error}");
                error_sender.send(()).await.unwrap() // this send should always be successful
            }
            Ok(_) => unreachable!("wait function should not return Ok()"),
        }
    });
```
If there will be error, yes, unwrap'ing it won't do much good for the logging. That's why we have `log::error!` above in case it fails.

Reply:
I still don't think this is the case. 
When node is stopped whilst farmer is working in CLI, farmer was giving a related error (`connection dropped` or something like that). 
When node stops, wouldn't farmer trigger the `Err(error) => {` branch of the `match`?

Reply:
Here is it: `2022-06-02T13:32:44.484948Z ERROR jsonrpsee_core::client::async_client: Error: Connection(Closed) terminating client`

Reply:
Hm, yes, you're right!

In that case I'm not sure you even need `match`, `if let Err(error) = multi_farming.wait().await` would be more compact.